lib/prism/translation/parser.rb

```diff
@@ -113,7 +113,13 @@ def unwrap(result, offset_cache)
           next unless valid_error?(error)
 
           location = build_range(error.location, offset_cache)
-          diagnostics.process(Diagnostic.new(error.message, location))
+          diagnostics.process(Diagnostic.new(error.message, :error, location))
+        end
+        result.warnings.each do |warning|
+          next unless valid_error?(warning)
```

Comment:
Can we introduce a `valid_warning?` here for this so that people can override that as well?

```suggestion
          next unless valid_warning?(warning)
```